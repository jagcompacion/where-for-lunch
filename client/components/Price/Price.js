import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Price extends Component {
  static propTypes = {
    price: PropTypes.arrayOf(PropTypes.number),
    onChange: PropTypes.func,
  };

  handleOnChangeAction = (e) => {
    const { value } = e.target;
    const { price, onChange } = this.props;
    const newPrice = price;
    const newValue = parseInt(value, 10);
    const index = price.indexOf(newValue);
    if (index > -1) newPrice.splice(index, 1);
    else newPrice.push(newValue);
    onChange(newPrice);
  }

  render() {
    const { price } = this.props;
    const options = ['$', '$$', '$$$', '$$$$'];
    return (
      <div>
        <div>price range:</div>
        <div>
          {options.map((item, index) => (
            <span key={index}>
              <input
                type="checkbox"
                value={item.length}
                onChange={this.handleOnChangeAction}
                checked={price && price.includes(item.length)}
              ></input>
              <span>{item}</span>
            </span>
          ))}
        </div>
      </div>
    );
  }
}
