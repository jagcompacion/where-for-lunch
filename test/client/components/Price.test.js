import React from 'react';
import { mount } from 'enzyme';
import Price from 'client/components/Price/Price';
import renderer from 'react-test-renderer';

test('With Enzyme, Price component will show 2 checkbox that are checked', () => {
  const price = [3, 4];
  const wrapper = mount(
    <Price price={price} />,
  );
  const b = wrapper.find({ type: 'checkbox' }).findWhere(n => n.props().checked);
  expect(b.length).toBe(2);
});

test('With Jest snapshot, Price component renders with no checkbox checked', () => {
  const b = renderer
    .create(<Price />)
    .toJSON();
  expect(b).toMatchSnapshot();
});
