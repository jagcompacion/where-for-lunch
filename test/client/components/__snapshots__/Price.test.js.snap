// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Price component renders with no checkbox checked 1`] = `
<div>
  <div>
    price range:
  </div>
  <div>
    <span>
      <input
        onChange={[Function]}
        type="checkbox"
        value={1}
      />
      <span>
        $
      </span>
    </span>
    <span>
      <input
        onChange={[Function]}
        type="checkbox"
        value={2}
      />
      <span>
        $$
      </span>
    </span>
    <span>
      <input
        onChange={[Function]}
        type="checkbox"
        value={3}
      />
      <span>
        $$$
      </span>
    </span>
    <span>
      <input
        onChange={[Function]}
        type="checkbox"
        value={4}
      />
      <span>
        $$$$
      </span>
    </span>
  </div>
</div>
`;
